lib/BiochemistryAPI/BiochemistryAPIImpl.py

```diff
@@ -277,18 +299,20 @@ def similarity_search(self, ctx, params):
         """
         # ctx is the context object
         # return variables are: matching_ids
-        #BEGIN similarity_search
+        # BEGIN similarity_search
         logging.info("Starting similarity_search")
         logging.info("Params: {}".format(params))
-        utils.check_param(params, ['query'], ['fp_type', 'min_similarity'])
-        params['structures'] = self.structures
+        utils.check_param(params, ["query"], ["fp_type", "min_similarity"])
```

Comment:
Is it obvious that when the `BiochemistryAPI.spec` says `fp_type` is either MACCS or Morgan, that Morgan must actually be RDKit? Maybe allow both Morgan and RDKit?

Reply:
This is another good catch, the `RDKFingerprint()` apparently doesn't use the Morgan approach, which would be the `GetMorganFingerprint()` function, but it takes an additional parameter, so the safer thing to do at this point is to update the spec itself, and leave a note that it *could* also do Morgan fingerprints if we extend it.